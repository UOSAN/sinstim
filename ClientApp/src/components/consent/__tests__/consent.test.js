import React from 'react';
import { shallow } from 'enzyme';

import Consent from '../consent';

it('should match snapshot', () => {
    const props = {
        onConsentAccept: () => {},
        onConsentDecline: () => {},
        text: 'consent text'
    };

    const wrapper = shallow(<Consent {...props} />);

    expect(wrapper).toMatchSnapshot();
});
