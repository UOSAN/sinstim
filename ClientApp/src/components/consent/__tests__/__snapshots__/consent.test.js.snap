// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Consent
    onSubmitConsent={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="consent-text"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam lacinia nunc elementum mauris porttitor, eu ultricies lectus maximus. Etiam in felis ut mauris porta consequat. Maecenas eget orci in purus tempor tincidunt id placerat justo. Suspendisse facilisis, lacus in lacinia varius, ex neque ornare mauris, eget molestie metus felis eget ligula. Mauris sem mi, pellentesque quis massa vitae, eleifend ullamcorper dolor. Donec sit amet faucibus nunc. Vivamus ipsum odio, pretium sit amet erat sagittis, pretium pulvinar libero. Curabitur eleifend non mi at cursus. Nam vitae ligula vel felis consectetur maximus ac quis nunc. Suspendisse potenti. Nam vestibulum massa non ante ornare consectetur.

Nulla nisi elit, lobortis et sagittis quis, pulvinar vel nisi. Nam dictum, nulla vel malesuada bibendum, neque nisl aliquet dolor, non rhoncus felis neque ut arcu. Sed suscipit iaculis ex ac fringilla. Nulla tincidunt magna vel nibh aliquam ornare. Curabitur tempus sem enim, at dapibus enim viverra nec. Mauris non porta lorem. Sed id est non lorem sagittis dignissim vel at est. Etiam orci tellus, dapibus elementum eros quis, pulvinar mattis arcu. Aliquam sollicitudin massa eu ipsum luctus, vitae eleifend purus vestibulum. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Sed lacinia erat at ligula facilisis, eu accumsan turpis luctus. Donec sollicitudin odio sit amet eros tempor, sit amet ullamcorper elit venenatis.
        </div>,
        <div
          className="consent-buttons"
        >
          <span
            className="consent-decline"
          >
            <button
              className="btn btn-primary"
              onClick={[Function]}
              type="button"
            >
              Decline
            </button>
          </span>
          <span
            className="consent-accept"
          >
            <button
              className="btn btn-primary"
              onClick={[Function]}
              type="button"
            >
              Accept
            </button>
          </span>
        </div>,
      ],
      "className": "consent",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam lacinia nunc elementum mauris porttitor, eu ultricies lectus maximus. Etiam in felis ut mauris porta consequat. Maecenas eget orci in purus tempor tincidunt id placerat justo. Suspendisse facilisis, lacus in lacinia varius, ex neque ornare mauris, eget molestie metus felis eget ligula. Mauris sem mi, pellentesque quis massa vitae, eleifend ullamcorper dolor. Donec sit amet faucibus nunc. Vivamus ipsum odio, pretium sit amet erat sagittis, pretium pulvinar libero. Curabitur eleifend non mi at cursus. Nam vitae ligula vel felis consectetur maximus ac quis nunc. Suspendisse potenti. Nam vestibulum massa non ante ornare consectetur.

Nulla nisi elit, lobortis et sagittis quis, pulvinar vel nisi. Nam dictum, nulla vel malesuada bibendum, neque nisl aliquet dolor, non rhoncus felis neque ut arcu. Sed suscipit iaculis ex ac fringilla. Nulla tincidunt magna vel nibh aliquam ornare. Curabitur tempus sem enim, at dapibus enim viverra nec. Mauris non porta lorem. Sed id est non lorem sagittis dignissim vel at est. Etiam orci tellus, dapibus elementum eros quis, pulvinar mattis arcu. Aliquam sollicitudin massa eu ipsum luctus, vitae eleifend purus vestibulum. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Sed lacinia erat at ligula facilisis, eu accumsan turpis luctus. Donec sollicitudin odio sit amet eros tempor, sit amet ullamcorper elit venenatis.",
          "className": "consent-text",
        },
        "ref": null,
        "rendered": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam lacinia nunc elementum mauris porttitor, eu ultricies lectus maximus. Etiam in felis ut mauris porta consequat. Maecenas eget orci in purus tempor tincidunt id placerat justo. Suspendisse facilisis, lacus in lacinia varius, ex neque ornare mauris, eget molestie metus felis eget ligula. Mauris sem mi, pellentesque quis massa vitae, eleifend ullamcorper dolor. Donec sit amet faucibus nunc. Vivamus ipsum odio, pretium sit amet erat sagittis, pretium pulvinar libero. Curabitur eleifend non mi at cursus. Nam vitae ligula vel felis consectetur maximus ac quis nunc. Suspendisse potenti. Nam vestibulum massa non ante ornare consectetur.

Nulla nisi elit, lobortis et sagittis quis, pulvinar vel nisi. Nam dictum, nulla vel malesuada bibendum, neque nisl aliquet dolor, non rhoncus felis neque ut arcu. Sed suscipit iaculis ex ac fringilla. Nulla tincidunt magna vel nibh aliquam ornare. Curabitur tempus sem enim, at dapibus enim viverra nec. Mauris non porta lorem. Sed id est non lorem sagittis dignissim vel at est. Etiam orci tellus, dapibus elementum eros quis, pulvinar mattis arcu. Aliquam sollicitudin massa eu ipsum luctus, vitae eleifend purus vestibulum. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Sed lacinia erat at ligula facilisis, eu accumsan turpis luctus. Donec sollicitudin odio sit amet eros tempor, sit amet ullamcorper elit venenatis.",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="consent-decline"
            >
              <button
                className="btn btn-primary"
                onClick={[Function]}
                type="button"
              >
                Decline
              </button>
            </span>,
            <span
              className="consent-accept"
            >
              <button
                className="btn btn-primary"
                onClick={[Function]}
                type="button"
              >
                Accept
              </button>
            </span>,
          ],
          "className": "consent-buttons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="btn btn-primary"
                onClick={[Function]}
                type="button"
              >
                Decline
              </button>,
              "className": "consent-decline",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Decline",
                "className": "btn btn-primary",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": "Decline",
              "type": "button",
            },
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="btn btn-primary"
                onClick={[Function]}
                type="button"
              >
                Accept
              </button>,
              "className": "consent-accept",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Accept",
                "className": "btn btn-primary",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": "Accept",
              "type": "button",
            },
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="consent-text"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam lacinia nunc elementum mauris porttitor, eu ultricies lectus maximus. Etiam in felis ut mauris porta consequat. Maecenas eget orci in purus tempor tincidunt id placerat justo. Suspendisse facilisis, lacus in lacinia varius, ex neque ornare mauris, eget molestie metus felis eget ligula. Mauris sem mi, pellentesque quis massa vitae, eleifend ullamcorper dolor. Donec sit amet faucibus nunc. Vivamus ipsum odio, pretium sit amet erat sagittis, pretium pulvinar libero. Curabitur eleifend non mi at cursus. Nam vitae ligula vel felis consectetur maximus ac quis nunc. Suspendisse potenti. Nam vestibulum massa non ante ornare consectetur.

Nulla nisi elit, lobortis et sagittis quis, pulvinar vel nisi. Nam dictum, nulla vel malesuada bibendum, neque nisl aliquet dolor, non rhoncus felis neque ut arcu. Sed suscipit iaculis ex ac fringilla. Nulla tincidunt magna vel nibh aliquam ornare. Curabitur tempus sem enim, at dapibus enim viverra nec. Mauris non porta lorem. Sed id est non lorem sagittis dignissim vel at est. Etiam orci tellus, dapibus elementum eros quis, pulvinar mattis arcu. Aliquam sollicitudin massa eu ipsum luctus, vitae eleifend purus vestibulum. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Sed lacinia erat at ligula facilisis, eu accumsan turpis luctus. Donec sollicitudin odio sit amet eros tempor, sit amet ullamcorper elit venenatis.
          </div>,
          <div
            className="consent-buttons"
          >
            <span
              className="consent-decline"
            >
              <button
                className="btn btn-primary"
                onClick={[Function]}
                type="button"
              >
                Decline
              </button>
            </span>
            <span
              className="consent-accept"
            >
              <button
                className="btn btn-primary"
                onClick={[Function]}
                type="button"
              >
                Accept
              </button>
            </span>
          </div>,
        ],
        "className": "consent",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam lacinia nunc elementum mauris porttitor, eu ultricies lectus maximus. Etiam in felis ut mauris porta consequat. Maecenas eget orci in purus tempor tincidunt id placerat justo. Suspendisse facilisis, lacus in lacinia varius, ex neque ornare mauris, eget molestie metus felis eget ligula. Mauris sem mi, pellentesque quis massa vitae, eleifend ullamcorper dolor. Donec sit amet faucibus nunc. Vivamus ipsum odio, pretium sit amet erat sagittis, pretium pulvinar libero. Curabitur eleifend non mi at cursus. Nam vitae ligula vel felis consectetur maximus ac quis nunc. Suspendisse potenti. Nam vestibulum massa non ante ornare consectetur.

Nulla nisi elit, lobortis et sagittis quis, pulvinar vel nisi. Nam dictum, nulla vel malesuada bibendum, neque nisl aliquet dolor, non rhoncus felis neque ut arcu. Sed suscipit iaculis ex ac fringilla. Nulla tincidunt magna vel nibh aliquam ornare. Curabitur tempus sem enim, at dapibus enim viverra nec. Mauris non porta lorem. Sed id est non lorem sagittis dignissim vel at est. Etiam orci tellus, dapibus elementum eros quis, pulvinar mattis arcu. Aliquam sollicitudin massa eu ipsum luctus, vitae eleifend purus vestibulum. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Sed lacinia erat at ligula facilisis, eu accumsan turpis luctus. Donec sollicitudin odio sit amet eros tempor, sit amet ullamcorper elit venenatis.",
            "className": "consent-text",
          },
          "ref": null,
          "rendered": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam lacinia nunc elementum mauris porttitor, eu ultricies lectus maximus. Etiam in felis ut mauris porta consequat. Maecenas eget orci in purus tempor tincidunt id placerat justo. Suspendisse facilisis, lacus in lacinia varius, ex neque ornare mauris, eget molestie metus felis eget ligula. Mauris sem mi, pellentesque quis massa vitae, eleifend ullamcorper dolor. Donec sit amet faucibus nunc. Vivamus ipsum odio, pretium sit amet erat sagittis, pretium pulvinar libero. Curabitur eleifend non mi at cursus. Nam vitae ligula vel felis consectetur maximus ac quis nunc. Suspendisse potenti. Nam vestibulum massa non ante ornare consectetur.

Nulla nisi elit, lobortis et sagittis quis, pulvinar vel nisi. Nam dictum, nulla vel malesuada bibendum, neque nisl aliquet dolor, non rhoncus felis neque ut arcu. Sed suscipit iaculis ex ac fringilla. Nulla tincidunt magna vel nibh aliquam ornare. Curabitur tempus sem enim, at dapibus enim viverra nec. Mauris non porta lorem. Sed id est non lorem sagittis dignissim vel at est. Etiam orci tellus, dapibus elementum eros quis, pulvinar mattis arcu. Aliquam sollicitudin massa eu ipsum luctus, vitae eleifend purus vestibulum. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Sed lacinia erat at ligula facilisis, eu accumsan turpis luctus. Donec sollicitudin odio sit amet eros tempor, sit amet ullamcorper elit venenatis.",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="consent-decline"
              >
                <button
                  className="btn btn-primary"
                  onClick={[Function]}
                  type="button"
                >
                  Decline
                </button>
              </span>,
              <span
                className="consent-accept"
              >
                <button
                  className="btn btn-primary"
                  onClick={[Function]}
                  type="button"
                >
                  Accept
                </button>
              </span>,
            ],
            "className": "consent-buttons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="btn btn-primary"
                  onClick={[Function]}
                  type="button"
                >
                  Decline
                </button>,
                "className": "consent-decline",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Decline",
                  "className": "btn btn-primary",
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": "Decline",
                "type": "button",
              },
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="btn btn-primary"
                  onClick={[Function]}
                  type="button"
                >
                  Accept
                </button>,
                "className": "consent-accept",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Accept",
                  "className": "btn btn-primary",
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": "Accept",
                "type": "button",
              },
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
